src/core/memory.lua

```diff
@@ -69,15 +69,15 @@ function reserve_new_page ()
    -- Check that we have permission
    lib.root_check("error: must run as root to allocate memory for DMA")
    -- Is the kernel shm limit too low for huge pages?
-   if huge_page_size > tonumber(syscall.sysctl("kernel.shmmax")) then
+   if huge_page_size > tonumber(lib.firstline("/proc/sys/kernel/shmmax")) then
       -- Yes: fix that
-      local old = syscall.sysctl("kernel.shmmax", tostring(huge_page_size))
+      local old = lib.writefile("/proc/sys/kernel/shmmax", tostring(huge_page_size))
       io.write("[memory: Enabling huge pages for shm: ",
                "sysctl kernel.shmmax ", old, " -> ", huge_page_size, "]\n")
    else
-      -- No: try provisioning an additional page
-      local have = tonumber(syscall.sysctl("vm.nr_hugepages"))
+      local have = tonumber(lib.firstline("/proc/sys/vm/nr_hugepages"))
       local want = have + 1
+      lib.writefile("/proc/sys/vm/nr_hugepages", tostring(want))
       syscall.sysctl("vm.nr_hugepages", tostring(want))
```

Comment:
Yes that was a mistake. Do I just push the removal change to the original
CR? Or is that something that you'll handle?

On 27 April 2016 at 13:58, Max Rottenkolber <notifications@github.com>
wrote:

> In src/core/memory.lua
> <https://github.com/snabbco/snabb/pull/896#discussion_r61251908>:
>
> >        local want = have + 1
> > +      lib.writefile("/proc/sys/vm/nr_hugepages", tostring(want))
> >        syscall.sysctl("vm.nr_hugepages", tostring(want))
>
> @petebristow <https://github.com/petebristow> ping. Pretty sure it was
> mistakenly leftover though.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/snabbco/snabb/pull/896/files/cbe97a4b68968c62825b2b74fd9486c1e27fbfc8..519311bcc79dd3bc41b016e9f9be4f383f397f01#r61251908>
>